src/map.rs

```diff
@@ -1473,15 +1475,16 @@ impl<K, V, S> Index<usize> for IndexMap<K, V, S> {
     /// Returns a reference to the value at the supplied `index`.
     ///
     /// ***Panics*** if `index` is out of bounds.
+    #[track_caller]
     fn index(&self, index: usize) -> &V {
-        self.get_index(index)
-            .unwrap_or_else(|| {
-                panic!(
-                    "index out of bounds: the len is {len} but the index is {index}",
-                    len = self.len()
-                );
-            })
-            .1
+        if let Some((_, value)) = self.get_index(index) {
+            value
+        } else {
+            panic!(
+                "index out of bounds: the len is {len} but the index is {index}",
+                len = self.len()
+            );
+        }
```

Comment:
I think you're right that this closure was in the way though, so this is a good change.